### DISCUSSION TITLE
Keep autoSelect behavior for a controlled Menu?

### DISCUSSION THREAD

**Discussion by sam-xif** [chakra-ui/chakra-ui#7047] [Q&A]
I am working with chakra for a UI, and I'm trying to add a hotkey which opens a menu component. Opening the menu normally (i.e. by clicking on it) auto selects the first element, which is desired, as it allows for arrow-key navigation up and down the menu. The initial solution I had in place to open the menu with the hotkey was to make the Menu controlled, and set `isOpen` to true in the hotkey handler, but the problem with this is that opening the menu in this way doesn't trigger the auto select behavior, so I need to use my mouse in order to select items on the menu. It looks like this is because the `useMenu` hook is what generates the `onOpen` callback that calls `focusFirstElement()`, but this callback does not get called when `isOpen` changes in a controlled way.

I'm not sure that this is a bug, per se, but perhaps there is a clean way to make this work? I have already tried using `ref.current.focus()` on one of the menu items, but that didn't really work properly, and I feel like using `dispatchEvent` to manually fire an onClick event is clunky as well.

Alternatively, I would propose adding as a behavior that `onOpen` is fired if `isOpen` changes from false to true, and potentially firing `onClose` if `isOpen` changes from true to false as well. That may get clunky though in the `onClose` case, because usually I use `onClose` handlers to set `isOpen` to false when the menu is controlled, so `isOpen` could be set to false redundantly in some cases. Another alternative might be to give access to onOpen in the body of the Menu (currently only `isOpen` and `onClose` are accessible)

What do you all think? Is there a cleaner solution for my problem that I haven't considered?

I'm using Chakra v1.8.6 for reference.

Check out a reproduction here: https://codesandbox.io/s/controlled-chakra-menu-eubqp2

**Comment by sam-xif** [comment]
Bump

**Comment by dbousamra** [comment]
I have same issue. Not got a workaround yet
